### DISCUSSION TITLE
Enhancement idea

### DISCUSSION THREAD

**Discussion by EHBradford** [rrirower/DP2SceneGraphWizard#56] [Ideas]
Hi, 

I host my json configuration file on one server that I control and my videos content on another which I have zero control or feedback.  Thus, I have no idea which videos are most watched.

Would it be possible to add a feature where when a video is played a quick http GET request (POST would be fine as well) is placed.  It would be very nice if this functionality passed along the video ID so that people who are in my shoes know which videos are being played.  Basically it would allow me to create metrics (most popular content creators, most popular videos, etc.) which in turn would allow me to cater my content better to my users.

Even a Rate up/Rate down onscreen control the user could choose would help.   As long as the feedback made its way back to an URL of my choice.  Of course with the video ID included as an argument.

Thank you.  I have successfully used your DP2SceneGraphWizard to create a new and better version of my channel.  Your work is greatly appreciated.

**Comment by ferdiworks** [comment]
@EHBradford Most video hosting plans come with analytics for videos. I myself have my own TV Channel software that creates the feed after you add your videos and create a playlist. We have built-in analytics in the software that we know what videos were played, and how many times, and in what region! So I understand what you are looking for. Even those using Vimeo or Wistia have analytics! The fact you are hosting the videos on your own is why you do not get analytics. I think even Amazon S3 gives some type of analytics. The DP to Scenegraph software is a great tool and I do not think that would be an easy task to get that coded in!

**Comment by rrirower** [comment]
@EHBradford First, please post your success story in [Success stories](https://github.com/rrirower/DP2SceneGraphWizard/discussions/55) discussion.  

Secondly, as @ferdiworks said, at the moment, the effort to incorporate what you want cannot be justified given the availability of other platforms that provide that service.  The changes you propose would affect all users.  Sadly, since you're the only one who has reported this as an enhancement, I can't justify the time at the moment given the current effort to release the code to support "subscriptions".

I will table your request for reconsideration.  I can't, however, promise it will be accommodated.

**Comment by EHBradford** [comment]
Okay,

It's your product.  It's not a problem.  I was hoping it would be one call, when the play button is pressed, to an external URL.  I was hoping roUrlTransfer would do the trick, but I'm not fully aware of how/what it does.   Regardless, the external call does not have to do anything other than pass an argument (the video ID).

As to the 5,000 video hosting platforms available, yes I am certain they offer that.  As I previously said, I do not host my files.  Therefore I do not get the benefits of a 3rd party service.  Thus, my need.  Hey free is free, you get what is given.  I got no problem with that.  :-)

I appreciate your timely answer to the question, have a GREAT DAY!   I'm off to learn Brightscript :-)

**Reply by 37media** [reply]
Just seeing this now, isn't this info available in your Roku Dashboard under  My Channels > View Analytics?

**Comment by rrirower** [comment]
@EHBradford beta4 now includes a new feature to record content voting.

**Reply by AndrewRCD** [reply]
Really cool what you added, have no idea who to create that API call thingy, but ... we all learn in time :) Good work @rrirower
